### PR TITLE
Gifup

### PR DESCRIPTION
**Pull request by BharatKalluri**
Github repository at https://github.com/BharatKalluri/Gifup

### CODE REVIEW

File: com.github.bharatkalluri.gifup.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
+    "app-id" : "com.github.bharatkalluri.gifup",
+    "runtime" : "org.gnome.Platform",
```
Comment (bilelmoussaoui) [inline com.github.bharatkalluri.gifup.json]: You could use elementary Loki's baseApp.

File: com.github.bharatkalluri.gifup.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    "runtime-version" : "master",
```
Comment (bilelmoussaoui) [inline com.github.bharatkalluri.gifup.json]: You shouldn't use the master version on flathub.

File: com.github.bharatkalluri.gifup.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/BharatKalluri/Gifup",
+                    "branch" : "master"
+                }
+            ]
+        }
+    ],
+    "build-options" : {
```
Comment (bilelmoussaoui) [inline com.github.bharatkalluri.gifup.json]: Could be removed

File: com.github.bharatkalluri.gifup.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                "/share/ffmpeg"
+            ]
+        },
+        {
+            "name" : "gifup",
+            "buildsystem" : "meson",
+            "builddir" : true,
+            "sources" : [
+                {
+                    "type" : "git",
```
Comment (bilelmoussaoui) [inline com.github.bharatkalluri.gifup.json]: Flathub only hosts reproducible builds. So you need either a release tarball or a specific commit here.

File: com.github.bharatkalluri.gifup.json [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "master",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "com.github.bharatkalluri.gifup",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--share=network",
```
Comment (bilelmoussaoui) [inline com.github.bharatkalluri.gifup.json]: For which reason the network is required?

File: com.github.bharatkalluri.gifup.json (hunk 4 above)
Comment (BharatKalluri) [reply com.github.bharatkalluri.gifup.json]: Added a commit hash now

File: com.github.bharatkalluri.gifup.json (hunk 2 above)
Comment (BharatKalluri) [reply com.github.bharatkalluri.gifup.json]: Shifted to 3.8

File: com.github.bharatkalluri.gifup.json (hunk 5 above)
Comment (BharatKalluri) [reply com.github.bharatkalluri.gifup.json]: This is not required, my bad.

File: com.github.bharatkalluri.gifup.json [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                "--disable-ffplay",
+                "--disable-ffserver",
+                "--disable-devices",
+                "--enable-gnutls",
+                "--enable-libvorbis"
+            ],
+            "sources" : [
+                {
+                    "type" : "archive",
+                    "url" : "https://ffmpeg.org/releases/ffmpeg-3.4.2.tar.xz",
```
Comment (TingPing) [inline com.github.bharatkalluri.gifup.json]: Is 4.0.2 a problem?

File: com.github.bharatkalluri.gifup.json [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                "install -Dm644 COPYING.LGPLv3 /app/share/licenses/ffmpeg/COPYING"
+            ],
+            "cleanup" : [
+                "/share/ffmpeg"
+            ]
+        },
+        {
+            "name" : "gifup",
+            "buildsystem" : "meson",
+            "builddir" : true,
```
Comment (TingPing) [inline com.github.bharatkalluri.gifup.json]: This can be removed, it defaults to `true` now.

File: com.github.bharatkalluri.gifup.json [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+            "post-install" : [
+                "install -Dm644 COPYING.LGPLv3 /app/share/licenses/ffmpeg/COPYING"
+            ],
+            "cleanup" : [
+                "/share/ffmpeg"
+            ]
+        },
+        {
+            "name" : "gifup",
+            "buildsystem" : "meson",
```
Comment (TingPing) [inline com.github.bharatkalluri.gifup.json]: You may want to pass
```json
"config-opts": [
  "--buildtype=release"
]
```
Or such.

**Comment by bilelmoussaoui** [comment]
The application doesn't seem to be working here for me
```
Gtk-Message: 16:52:38.460: Failed to load module "pk-gtk-module"
Gtk-Message: 16:52:38.465: Failed to load module "pk-gtk-module"
Splitting the commandline.
Reading option '-hide_banner' ... matched as option 'hide_banner' (do not show program banner) with argument '1'.
Reading option '-loglevel' ... matched as option 'loglevel' (set logging level) with argument 'debug'.
Reading option '-ss' ... matched as option 'ss' (set the start time offset) with argument '1'.
Reading option '-i' ... matched as input url with argument '///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4'.
Reading option '-frames:v' ... matched as option 'frames' (set the number of frames to output) with argument '1'.
Reading option '-filter:v' ... matched as option 'filter' (set stream filtergraph) with argument 'scale=150:-1'.
Reading option '/tmp/gifup_start.bmp' ... matched as output url.
Reading option '-y' ... matched as option 'y' (overwrite output files) with argument '1'.
Finished splitting the commandline.
Parsing a group of options: global .
Applying option hide_banner (do not show program banner) with argument 1.
Applying option loglevel (set logging level) with argument debug.
Applying option y (overwrite output files) with argument 1.
Successfully parsed a group of options.
Parsing a group of options: input url ///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4.
Applying option ss (set the start time offset) with argument 1.
Successfully parsed a group of options.
Opening an input file: ///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4.
[NULL @ 0x10eb480] Opening '///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4' for reading
[file @ 0x10ebe00] Setting default whitelist 'file,crypto'
///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4: No such file or directory

** (com.github.bharatkalluri.gifup:2): CRITICAL **: 16:52:43.588: Utils.vala:120: Child process exited with code 1
Gtk-Message: 16:52:43.601: GtkDialog mapped without a transient parent. This is discouraged.
Splitting the commandline.
Reading option '-hide_banner' ... matched as option 'hide_banner' (do not show program banner) with argument '1'.
Reading option '-loglevel' ... matched as option 'loglevel' (set logging level) with argument 'debug'.
Reading option '-ss' ... matched as option 'ss' (set the start time offset) with argument '2'.
Reading option '-i' ... matched as input url with argument '///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4'.
Reading option '-frames:v' ... matched as option 'frames' (set the number of frames to output) with argument '1'.
Reading option '-filter:v' ... matched as option 'filter' (set stream filtergraph) with argument 'scale=150:-1'.
Reading option '/tmp/gifup_start.bmp' ... matched as output url.
Reading option '-y' ... matched as option 'y' (overwrite output files) with argument '1'.
Finished splitting the commandline.
Parsing a group of options: global .
Applying option hide_banner (do not show program banner) with argument 1.
Applying option loglevel (set logging level) with argument debug.
Applying option y (overwrite output files) with argument 1.
Successfully parsed a group of options.
Parsing a group of options: input url ///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4.
Applying option ss (set the start time offset) with argument 2.
Successfully parsed a group of options.
Opening an input file: ///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4.
[NULL @ 0xdb3480] Opening '///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4' for reading
[file @ 0xdb3e00] Setting default whitelist 'file,crypto'
///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4: No such file or directory

** (com.github.bharatkalluri.gifup:2): CRITICAL **: 16:52:45.154: Utils.vala:120: Child process exited with code 1
Gtk-Message: 16:52:45.161: GtkDialog mapped without a transient parent. This is discouraged.
Splitting the commandline.
Reading option '-hide_banner' ... matched as option 'hide_banner' (do not show program banner) with argument '1'.
Reading option '-loglevel' ... matched as option 'loglevel' (set logging level) with argument 'debug'.
Reading option '-ss' ... matched as option 'ss' (set the start time offset) with argument '3'.
Reading option '-i' ... matched as input url with argument '///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4'.
Reading option '-frames:v' ... matched as option 'frames' (set the number of frames to output) with argument '1'.
Reading option '-filter:v' ... matched as option 'filter' (set stream filtergraph) with argument 'scale=150:-1'.
Reading option '/tmp/gifup_start.bmp' ... matched as output url.
Reading option '-y' ... matched as option 'y' (overwrite output files) with argument '1'.
Finished splitting the commandline.
Parsing a group of options: global .
Applying option hide_banner (do not show program banner) with argument 1.
Applying option loglevel (set logging level) with argument debug.
Applying option y (overwrite output files) with argument 1.
Successfully parsed a group of options.
Parsing a group of options: input url ///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4.
Applying option ss (set the start time offset) with argument 3.
Successfully parsed a group of options.
Opening an input file: ///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4.
[NULL @ 0xeda480] Opening '///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4' for reading
[file @ 0xedae00] Setting default whitelist 'file,crypto'
///home/belmoussaoui/Videos/WordPress/WordPress Plugin %2345 - Quick SCSS and JS form styling-OCXqWjO1L34.mp4: No such file or directory

** (com.github.bharatkalluri.gifup:2): CRITICAL **: 16:52:46.481: Utils.vala:120: Child process exited with code 1
Gtk-Message: 16:52:46.487: GtkDialog mapped without a transient parent. This is discouraged.

(com.github.bharatkalluri.gifup:2): GLib-GObject-CRITICAL **: 16:52:52.197: g_object_ref_sink: assertion 'G_IS_OBJECT (object)' failed

(com.github.bharatkalluri.gifup:2): GLib-GObject-CRITICAL **: 16:52:52.231: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
```

**Comment by BharatKalluri** [comment]
There is some issue with flatpak, It'll take some time to debug. I will close this pull request and reopen It once I fix it. 

Thanks!

**Pull request closed without merging**